frameworks/helloworld/tests/test_placement.py

```diff
@@ -73,6 +73,153 @@ def test_rack_not_found():
     sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
 
 
+@pytest.mark.dcos_min_version('1.11')
+@pytest.mark.sanity
+def test_unique_zone_fails():
+    """Create a new user.
+    This test assumes that the DC/OS cluster is reportin that all agents are in a single zone.
+    """
+    options = {
+        'service': {
+            'spec_file': 'examples/marathon_constraint.yml'
+        },
+        'hello': {
+            'placement': '@zone:UNIQUE'
+        },
+        'world': {
+            'placement': '@zone:UNIQUE'
+        }
+    }
+
+    fail_placement(options)
+
+
+@pytest.mark.dcos_min_version('1.11')
+@pytest.mark.sanity
+def test_max_per_zone_fails():
+    """Create a new user.
+    This test assumes that the DC/OS cluster is reportin that all agents are in a single zone.
+    """
+    options = {
+        'service': {
+            'spec_file': 'examples/marathon_constraint.yml'
+        },
+        'hello': {
+            'placement': '@zone:MAX_PER:1'
+        },
+        'world': {
+            'placement': '@zone:MAX_PER:1'
+        }
+    }
+
+    fail_placement(options)
+
+
+@pytest.mark.dcos_min_version('1.11')
+@pytest.mark.sanity
+def test_max_per_zone_succeeds():
+    """Create a new user.
+    This test assumes that the DC/OS cluster is reportin that all agents are in a single zone.
+    """
+    options = {
+        'service': {
+            'spec_file': 'examples/marathon_constraint.yml'
+        },
+        'hello': {
+            'placement': '@zone:MAX_PER:1'
+        },
+        'world': {
+            'placement': '@zone:MAX_PER:2'
+        }
+    }
+
+    succeed_placement(options)
+
+
+@pytest.mark.dcos_min_version('1.11')
+@pytest.mark.sanity
+def test_group_by_zone_succeeds():
+    options = {
+        'service': {
+            'spec_file': 'examples/marathon_constraint.yml'
+        },
+        'hello': {
+            'placement': '@zone:GROUP_BY:1'
+        },
+        'world': {
+            'placement': '@zone:GROUP_BY:1'
+        }
+    }
+    succeed_placement(options)
+
+
+@pytest.mark.dcos_min_version('1.11')
+@pytest.mark.sanity
+def test_group_by_zone_fails():
+    options = {
+        'service': {
+            'spec_file': 'examples/marathon_constraint.yml'
+        },
+        'hello': {
+            'placement': '@zone:GROUP_BY:1'
+        },
+        'world': {
+            'placement': '@zone:GROUP_BY:2'
+        }
+    }
+
+    fail_placement(options)
+
+
+def succeed_placement(options):
+    """
```

Comment:
These docs look wrong (here and below), copied from one of the test cases?

Reply:
These docs are correct.

Reply:
Oh I get it now.